script/clang-format

```
@@ -13,11 +13,43 @@ import sys
 import threading
 
 
+
+def get_binary(name, version):
+    clang_binary = f"{name}-{version}"
+    try:
+        result = subprocess.check_output([clang_binary, '-version'])
+        if result.returncode == 0:
+            return clang_binary
+    except Exception:
+        pass
+    clang_binary = name
+    try:
+        result = subprocess.run([clang_binary, '-version'], text=True, capture_output=True)
+        if result.returncode == 0 and (f"version {version}") in result.stdout:
+            return clang_binary
+    except Exception as ex:
+        pass
+
+    print(
+        f"""
+        Oops. It looks like {name} is not installed.
+
+        Please confirm you can run "{name} -version" or "{name}-{version} -version"
+        in your terminal and install
+        {name} (v{version}) if necessary.
+
+        Note you can also upload your code as a pull request on GitHub and see the CI check
+        output to apply {name}
+        """
+    )
+    sys.exit()
```

Ideally functions don't ever call `sys.exit()` as it makes tracing execution flow more difficult. It would be better to let this throw and catch it

---

Also please add how to install. Just for dummy's like me, who get puzzled how and where something need to be done.

---

> Ideally functions other than `main` don't ever call `sys.exit()` as it makes tracing execution flow more difficult. It would be better to let this throw and catch it

I didn't change that. Not trying to refactor the whole thing, just wanted to make it run.

---

> Also please add how to install. Just for dummy's like me, who get puzzled how and where something need to be done.

It's installed in `venv/bin` by `script/setup` and then you should have done `source venv/bin/activate` which will put that in your PATH.

---

Well, it *wasn't* being installed. It is now.